src/main/webapp/static/js/web3provider.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+
+// Unpkg imports
+const Web3Modal = window.Web3Modal.default;
+const WalletConnectProvider = window.WalletConnectProvider.default;
+const Fortmatic = window.Fortmatic;
+const evmChains = window.evmChains;
+
+// Web3modal instance
+let web3Modal;
+
+// Chosen wallet provider given by the dialog window
+let provider;
+
+
+/**
+ * Setup the orchestra
+ */
+function init() {
+
+  console.log("WalletConnectProvider is", WalletConnectProvider);
+  console.log("Fortmatic is", Fortmatic);
+  console.log("window.web3 is", window.web3, "window.ethereum is", window.ethereum);
+
+  // Check that the web page is run in a secure context,
+  // as otherwise MetaMask won't be available
+  if(location.protocol !== 'https:') {
+    // https://ethereum.stackexchange.com/a/62217/620
+    // TODO:
+  }
+
+  // Tell Web3modal what providers we have available.
+  // Built-in web browser provider (only one can exist as a time)
+  // like MetaMask, Brave or Opera is added automatically by Web3modal
+  const providerOptions = {
+    walletconnect: {
+      package: WalletConnectProvider,
+      options: {
+        // Mikko's test key - don't copy as your mileage may vary
+        infuraId: "8043bb2cf99347b1bfadfb233c5325c0",
+      }
+    },
+
+    fortmatic: {
+      package: Fortmatic,
+      options: {
+        // Mikko's TESTNET api key
+        key: "pk_test_391E26A3B43A3350"
```

Comment:
We need to generate our own Fortmatic key?

Reply:
If we want to support Fortmatic, then yes!

Reply:
I suggest we skip Fortmatic, if you are okay with that.

Reply:
Sure, np.